govcd/orgvdcnetwork.go

```diff
@@ -119,23 +157,19 @@ func (vdc *Vdc) CreateOrgVDCNetwork(networkConfig *types.OrgVDCNetwork) error {
 						time.Sleep(3 * time.Second)
 						continue
 					}
-					return fmt.Errorf("error instantiating a new OrgVDCNetwork: %s", err)
+					return Task{}, fmt.Errorf("error instantiating a new OrgVDCNetwork: %s", err)
 				}
 				break
 			}
-			newstuff := NewOrgVDCNetwork(vdc.client)
-			if err = decodeBody(resp, newstuff.OrgVDCNetwork); err != nil {
-				return fmt.Errorf("error decoding orgvdcnetwork response: %s", err)
+			orgVDCNetwork := NewOrgVDCNetwork(vdc.client)
+			if err = decodeBody(resp, orgVDCNetwork.OrgVDCNetwork); err != nil {
+				return Task{}, fmt.Errorf("error decoding orgvdcnetwork response: %s", err)
 			}
-			task := NewTask(vdc.client)
-			for _, taskItem := range newstuff.OrgVDCNetwork.Tasks.Task {
-				task.Task = taskItem
-				err = task.WaitTaskCompletion()
-				if err != nil {
-					return fmt.Errorf("Error performing task: %#v", err)
-				}
+			for _, taskItem := range orgVDCNetwork.OrgVDCNetwork.Tasks.Task {
```

Comment:
I am not sure this is safe, cause there can be running previous task and we are not sure that first one is ours.

Reply:
This one should be safe, as the network is the one we have just created, and there should not be any other task running. If there is, the whole operation would become meaningless, because we can only give back one task.
I will add a check to make sure there is only one active task, though.